src/figuresClasses.ts

```diff
@@ -1,19 +1,70 @@
+type Fig = 'triangle' | 'circle' | 'rectangle';
+type Col = 'red' | 'green' | 'blue';
+
+const down: (num: number) => number = num =>
+  Math.floor(num * 100) / 100;
+
 export interface Figure {
+  shape: Fig;
+  color: Col;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Fig = 'triangle';
+
+  constructor(
+    public color: Col,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0
+        || this.a + this.b <= this.c) {
+      throw new Error('not a valid triangle');
+    }
+  }
+
+  getArea(): number {
+    const p: number = (this.a + this.b + this.c) / 2;
```

Comment:
не указывай примитивы, они вычисляются на основе значения
если ты записываешь число, автоматически подставится тип число
```suggestion
    const p = (this.a + this.b + this.c) / 2;
```